sc/src/test/java/uk/ac/ebi/atlas/experimentpage/JsonExperimentTSnePlotControllerWIT.java

```diff
@@ -164,4 +163,29 @@ void validJsonForInvalidK() throws Exception {
                 .andExpect(content().contentType(MediaType.APPLICATION_JSON_UTF8))
                 .andExpect(jsonPath("$.series", hasSize(1)));
     }
+
+    @Test
+    void validJsonWithValidExperimentAccession() throws Exception {
+        String [] experimentAccession = {"E-MTAB-5061", "E-EHCA-2"};
```

Comment:
Try to use either `jdbcTestUtils` to fetch random accessions, or use parameterised tests to exhaustively test experiments (see for example `CellMetadataServiceIT`).

Reply:
I was using that approach first but issues arise when it comes to experiment with no metadata. So have to hard code to experiments that have metadata to test.

Reply:
You could make use of parameterized tests and provider methods, see an example of this here: https://github.com/ebi-gene-expression-group/atlas/blob/dev/sc/src/test/java/uk/ac/ebi/atlas/metadata/JsonCellMetadataControllerWIT.java#L111

Reply:
I wasn't aware of this functionality to filter accessions. Will fix and update code.